californium-core/src/main/java/org/eclipse/californium/core/network/Exchange.java

```diff
@@ -748,23 +749,21 @@ public static KeyMID fromOutboundMessage(Message message) {
 	 */
 	public static final class KeyToken {
 
-		private static final int MAX_PORT_NO = (1 << 16) - 1;
 		private final byte[] token;
-		private final byte[] address;
-		private final int port;
+		private final byte[] identity;
 		private final int hash;
 
-		private KeyToken(byte[] token, byte[] address, int port) {
+		private KeyToken(final byte[] token, final byte[] identity) {
```

Comment:
Details : I don't really understand the benefits of using final keyword for parameters. (this make the signature heavier)